File: week05/2/Prob2.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* now = head;
        ListNode* prev = nullptr;

        while(now){
            if(prev != nullptr && now->val == prev->val){
                prev->next = now->next;
            }else{
                prev = now;
            }
            now = now->next;
        }
        return head;
    }
};

File: week05/2/Prob2.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        now = head
        prev = None
        while now:
            chkval = now.val
            if prev != None and prev.val == now.val:
                prev.next = now.next
            else:
                prev = now
            now = now.next
        
        return head

